cmd/syncer/cmd/syncer.go

```diff
@@ -0,0 +1,164 @@
+/*
+Copyright 2022 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"errors"
+	"fmt"
+
+	nscontroller "github.com/kcp-dev/kcp/pkg/reconciler/namespace"
+	"github.com/kcp-dev/kcp/pkg/syncer"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	"k8s.io/apimachinery/pkg/util/sets"
+	genericapiserver "k8s.io/apiserver/pkg/server"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/component-base/logs"
+	"k8s.io/klog/v2"
+)
+
+const numThreads = 2
+
+func NewSyncerCommand() *cobra.Command {
+	options := NewDefaultOptions()
+
+	syncerCommand := &cobra.Command{
+		Use:   "syncer",
+		Short: "Synchronizes resources in `kcp` assigned to the clusters",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			options.SyncedResourceTypes = args
+
+			//validation
+			err := options.Validate()
+			if err != nil {
+				return err
+			}
+
+			return nil
+		},
+	}
+
+	//bind flags
+	options.BindFlags(syncerCommand.Flags())
+
+	return syncerCommand
+}
+
+func (options *Options) Validate() error {
+	err := options.Complete()
+	if err != nil {
+		return err
+	}
+
+	err = options.Run()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (options *Options) Complete() error {
+	if len(options.SyncedResourceTypes) == 0 {
+		options.SyncedResourceTypes = []string{"deployments.apps"}
+		klog.Infoln("No resource types provided; using defaults.")
+	}
```

Comment:
could we have that as default list in the Options struct on construction? Or is this a problem that a string set flag then cannot override it?

Reply:
Hey @ncdc , by "change", do you mean making ```SyncedResourceTypes``` a flag (StringArrayVar type)? I think that could be done.

Reply:
@RAMESSESII2 that's right, we could make SyncedResourceTypes a flag.